### PR TITLE
fix filter disk Index Search (filter threshold build unverified)

### PR DESCRIPTION
**Pull request by NeelamMahapatro**
- [ ] Does this PR have a descriptive title that could go in our release notes?
- [ ] Does this PR add any new dependencies?
- [ ] Does this PR modify any existing APIs?
   - [ ] Is the change to the API backwards compatible?
- [ ] Should this result in any changes to our documentation, either updating existing docs or adding new ones?
 
#### What does this implement/fix? Briefly explain your changes.

Fix filter search for memory and disk Index

- fixed label file format to be in sync (while reading the format was [ filters ] however during writing format was [ id \t filters ]). hence Invalid labels were getting used during search
- During Disk Index load, the argument for parse_label_file was incorrect (undeclared variable disk_index_file). hence, label file was not getting loaded.
- Resized query_result_dist vector for memory index
- gen_random_slice uses augmented data in case of filter threshold > 0. But the augmented file was deleted before generating the slices. Hence, shifted the remove file part after random gen_random_slice function.

filtered build and search disk and memory index verified with filter threshold 0.

### CODE REVIEW

**Comment by harsha-simhadri** [comment]
@NeelamMahapatro  could you please edit the PR description to include what issue your PR fixes and what changes are made for the fix. Please also take a look at the check list to mark if any API has changed.

File: src/disk_utils.cpp [hunk 1]
```diff
@@ -1196,19 +1196,19 @@ namespace diskann {
     // into replica dummy points which evenly distribute the filters. The rest
     // of index build happens on the augmented base and labels
     std::string augmented_data_file, augmented_labels_file;
-    if (use_filters) {
+    if (use_filters) {
```
Comment (harsha-simhadri) [inline src/disk_utils.cpp:1199]: Did you run clang format before pushing?

File: src/index.cpp [hunk 2]
```diff
@@ -292,7 +292,6 @@ namespace diskann {
         if (_pts_to_labels.size() > 0) {
           std::ofstream label_writer(std::string(filename) + "_labels.txt");
           for (_u32 i = 0; i < _pts_to_labels.size(); i++) {
```
Comment (harsha-simhadri) [inline src/index.cpp:294]: So what is the new file format for filters? Could you please document this as comments somewhere in the code?

Optionally (not required for closing this PR) Consider writing a worksflow for building, computing GT, and searching filtered index in the ROOT/workflows/ directory. You could use the workflow guides there as examples.

File: src/pq_flash_index.cpp [hunk 3]
```diff
@@ -559,9 +559,15 @@ namespace diskann {
   template<typename T>
   int PQFlashIndex<T>::load(uint32_t num_threads, const char *index_prefix) {
 #endif
+    std::string pq_table_bin = std::string(index_prefix) + "_pq_pivots.bin";
```
Comment (harsha-simhadri) [inline src/pq_flash_index.cpp:562]: again, did you run clang format on this?

**Review by harsha-simhadri — CHANGES_REQUESTED** [review]
Please check if you have run clang format before committing your changes. If not, please do and push formatted code.
 Would be good to get @dkaczynski sign off since he is worked on this branch. 
Otherwise, I am OK with the merge.

File: src/index.cpp (hunk 2 above)
Comment (NeelamMahapatro) [reply src/index.cpp:294]: the label file format I have mentioned in the comments. 
Workflows - will be including in a separate PR

File: src/index.cpp [hunk 4]
```diff
@@ -1667,6 +1666,9 @@ namespace diskann {
 
 template<typename T, typename TagT>
   void Index<T, TagT>::parse_label_file(const std::string &map_file) {
+    // old Format of Label txt file: id \t filters with comma separators
+    // new Format of Label txt file: filters with comma separators
```
Comment (harsha-simhadri) [inline src/index.cpp:1670]: Lets just say what the format after PR would be and leave out the old format.

**Pull request merged**